packages/shared/src/components/layout/MainLayoutHeader.tsx

```diff
@@ -77,23 +86,44 @@ function MainLayoutHeader({
     campaignKey: ReferralCampaignKey.Search,
   });
 
+  const { currentStreak } = useReadingStreak();
+
   const renderButtons = () => {
     return (
       <>
-        <CreatePostButton
-          className={classNames(
-            'mr-0 laptop:mr-3',
-            optOutWeeklyGoal ? 'tablet:mr-0' : 'tablet:mr-3',
-          )}
-        />
+        <CreatePostButton />
+        {user && readingStreak === ReadingStreaksExperiment.V1 && (
```

Comment:
For now this is fine, but it will get a custom hook I guess?

Reply:
Yeah, if we ever need additional logic to determine the result and keep everything inline.

Reply:
should we bite the bullet and add a hook now? We will need to make this check in other places as well, so might as well have a hook we can go to make changes if we ever need to in the future

Reply:
Yeah, that makes sense. Would be easier to update everything down the line.